core/src/main/java/org/bitcoinj/core/Transaction.java

```diff
@@ -693,9 +693,10 @@ public String toString(@Nullable AbstractBlockChain chain) {
             }
             s.append(String.format(Locale.US, "%n"));
         }
-        Coin fee = getFee();
+        final Coin fee = getFee();
         if (fee != null)
-            s.append("     fee  ").append(fee.toFriendlyString()).append(String.format(Locale.US, "%n"));
+            s.append("     fee  ").append(fee.multiply(1000).divide(unsafeBitcoinSerialize().length).toFriendlyString())
```

Comment:
@jlopp My point is that per byte has not enough precision, as the number of satoshis per byte would be in the 10-50 range (and even has been lower in the past). Unless you print fractions of a satoshi, but I'm not sure this is what you want.